services/provider/server/storageclaim.go

```diff
@@ -98,19 +98,49 @@ func (s *storageClassClaimManager) Create(
 		},
 	})
 
-	err = s.client.Create(ctx, storageClassClaimObj)
+	claimBytes, _ := json.Marshal(storageClassClaimObj)
+	secretObj := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				ocsv1alpha1.StorageClassClaimAnnotation: fmt.Sprintf("%s/%s", storageClassClaimObj.Namespace, storageClassClaimObj.Name),
+			},
+			Name:      fmt.Sprintf("%s-claim", storageClassClaimObj.Name),
+			Namespace: s.namespace,
+			Labels:    storageClassClaimObj.ObjectMeta.Labels,
+		},
+		Data: map[string][]byte{
+			"StorageClassClaim": claimBytes,
+		},
+	}
+	secretObj.SetOwnerReferences([]metav1.OwnerReference{
+		{
+			APIVersion:         gvk.GroupVersion().String(),
+			Kind:               gvk.Kind,
+			UID:                consumer.GetUID(),
+			Name:               consumer.GetName(),
+			BlockOwnerDeletion: pointer.BoolPtr(true),
+		},
+	})
+
+	err = s.client.Create(ctx, secretObj)
 	if err != nil {
 		if !kerrors.IsAlreadyExists(err) {
-			return fmt.Errorf("failed to create a StorageClassClaim named %q for consumer %q and claim %q. %w", generatedClaimName, consumerUUID, storageClassClaimName, err)
+			return fmt.Errorf("failed to create a Secret named %q for consumer %q and claim %q. %w", generatedClaimName, consumerUUID, storageClassClaimName, err)
 		}
-		newStorageClassClaimObj := &ocsv1alpha1.StorageClassClaim{}
-		getErr := s.client.Get(ctx, client.ObjectKey{Name: generatedClaimName, Namespace: s.namespace}, newStorageClassClaimObj)
+		existingSecret := &corev1.Secret{}
+		getErr := s.client.Get(ctx, client.ObjectKey{Name: secretObj.Name, Namespace: s.namespace}, existingSecret)
 		if getErr != nil {
-			klog.Errorf("failed to get a StorageClassClaim named %q for consumer %q and claim %q. %v", generatedClaimName, consumerUUID, storageClassClaimName, getErr)
+			klog.Errorf("failed to get a Secret named %q for consumer %q and claim %q. %v", generatedClaimName, consumerUUID, storageClassClaimName, getErr)
+			return err
+		}
+		newStorageClassClaimObj := &ocsv1alpha1.StorageClassClaim{}
+		unmarshallErr := json.Unmarshal(existingSecret.Data["StorageClassClaim"], newStorageClassClaimObj)
+		if unmarshallErr != nil {
+			klog.Errorf("failed to unmarshall the StorageClassClaim named %q for consumer %q and claim %q. %v", generatedClaimName, consumerUUID, storageClassClaimName, unmarshallErr)
```

Comment:
Why do we need to unmarshal here?

Reply:
in line 148 we are comparing existing and requested spec change

Reply:
In that case, let's move the unmarshal just above that section of the code.

Reply:
This was to keep the comparison strictly with the Spec as before, otherwise the comparison would also include the Status changes (and Meta, but that should never relaly change).